### PR TITLE
dep: add asdf to make ql-deps; git pull if dir

### PR DESCRIPTION
**Pull request by WillForan**
For #58, add $(QLDIR)/adsf recipe

Also added make function to pull instead of clone if local-projects dir already exists

### CODE REVIEW

**Comment by vindarel** [comment]
This would pull this version of asdf for all users who run the install command, regardless of their existing asdf version (if any). I wonder if this can be an issue.

WDYT everybody?

Overall very welcome improvements that I'll merge in, thank you.

File: Makefile [hunk 1]
```diff
@@ -2,38 +2,57 @@ LISP ?= sbcl
 
 all: build
 
+# can use bespoke dir like 'QLDIR=~/nostandard/local-projects make'
+QLDIR ?= $(HOME)/quicklisp/local-projects
```
Comment (vindarel) [inline Makefile:6]: this is not exactly the QL dir but its local-projects directory, I'd change sthg to be more precise.

**Comment by tmtvl** [comment]
> This would pull this version of asdf for all users who run the install command, regardless of their existing asdf version (if any). I wonder if this can be an issue.

Yeah, trying to do dependency management in pure Makefile can be a bit of a pain in the patookus.

Actually, it may be fine to just do it like this and to have a Makefile.lisp file which does most of the same stuff using UIOP, which would let us do the ASDF version check. That would allow people who have ASDF to use the lisp file and people without it to use this standard Makefile.

Or we could just make a smaller lisp script which just outputs a 1 or a 0 depending on whether if a sufficiently up-to-date version of ASDF is installed.

**Comment by vindarel** [comment]
the script can be:

```lisp
;; sbcl --script check-asdf-version.lisp
(require 'asdf)

(uiop:format! t "asdf version: ~a~&" (asdf:asdf-version))
(let ((version (asdf:asdf-version)))
  (cond
    ((uiop:version<= version "3.3.4")
     (uiop:format! *error-output* "your asdf version is too old.~&")
     (uiop:quit 1))
    (t
     (uiop:quit 0))))
```
*(and oh, without dependencies, stock sbcl scripts run acceptably fast too. 0.10s)*

File: Makefile [hunk 2]
```diff
@@ -2,38 +2,57 @@ LISP ?= sbcl
...
+	# https://github.com/ciel-lang/CIEL/issues/58
+	mkdir -p $(QLDIR)
+	cd $(QLDIR) && \
+		curl -sL https://asdf.common-lisp.dev/archives/asdf-3.3.5.tar.gz | \
+		tar -xvzf - && \
+		mv asdf-3.3.5 asdf
+
 # Install some Quicklisp dependencies.
-ql-deps:
+ql-deps: $(QLDIR)/asdf
```
Comment (vindarel) [inline Makefile:24]: I suggest that either we remove this automatic target dependency, either we check the asdf version. (I'll accept both, preference for the check).

Comment (vindarel) [reply Makefile:24]: I removed the automatic download and added a version check.

**Comment by vindarel** [comment]
@WillForan Thanks for the well useful PR.

**Comment by WillForan** [comment]
ahh sorry I was too slow to help clean it up!

**Pull request merged**